nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/state/controller-services/controller-services.effects.ts

```diff
@@ -635,6 +636,82 @@ export class ControllerServicesEffects {
         { dispatch: false }
     );
 
+    openMoveControllerServiceDialog$ = createEffect(
+        () =>
+            this.actions$.pipe(
+                ofType(ControllerServicesActions.openMoveControllerServiceDialog),
+                map((action) => action.request),
+                concatLatestFrom(() => this.store.select(selectCurrentProcessGroupId)),
+                tap(([request, currentProcessGroupId]) => {
+                    const serviceId: string = request.id;
+                    request.processGroupFlow = this.flowService.getFlow(currentProcessGroupId);
```

Comment:
The state for the `flow-designer` page is already loaded with the current process group. We should be able to simply use that instead of making another request here.

Reply:
The current state did not store the entire flow. I did change this and I was able to use it here without an api call. However, with the implementation of disabled drop down menu process groups, I did need to 2 other api calls here.

Reply:
Ah sorry. Yes you are correct. I thought we had stashed the entire `processGroupFlow` in the service state as well but I see that only saved the breadcrumbs and parameter context details. Now we're saving the entire `processGroupFlow` next to the breadcrumbs and parameter context details.

Do we only need the `processGroupFlow` in order to get the child Process Groups for the Move dialog? If so, what are your thoughts only saving them in the store. Often times the flow can be very large and in this page we won't need most of those details.

Let me know if I missed something and we do need additional details.

Reply:
Thank you for the idea, I hadn't thought of that. I removed the storing of the whole process group flow, and just saved an array with names and IDs.